interfaces/apparmor/apparmor.go

```diff
@@ -73,24 +64,36 @@ func loadProfile(fname, cacheDir string, flags aaParserFlags) error {
 	if !osutil.GetenvBool("SNAPD_DEBUG") {
 		args = append(args, "--quiet")
 	}
-	args = append(args, fname)
+	args = append(args, fnames...)
 
 	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("cannot load apparmor profile: %s\napparmor_parser output:\n%s", err, string(output))
+		return fmt.Errorf("cannot load apparmor profiles: %s\napparmor_parser output:\n%s", err, string(output))
 	}
 	return nil
 }
 
-func unloadProfile(name, cacheDir string) error {
-	output, err := exec.Command("apparmor_parser", "--remove", name).CombinedOutput()
+// UnloadProfiles removes the named profiles from the running kernel.
+//
+// The operation is done with: apparmor_parser --remove $names...
+// The binary cache file is removed from /var/cache/apparmor
+func unloadProfiles(names []string, cacheDir string) error {
+	if len(names) == 0 {
+		return nil
+	}
+
+	args := []string{"--remove"}
+	args = append(args, names...)
+	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
 	if err != nil {
 		return fmt.Errorf("cannot unload apparmor profile: %s\napparmor_parser output:\n%s", err, string(output))
 	}
-	err = os.Remove(filepath.Join(cacheDir, name))
-	// It is not an error if the cache file wasn't there to remove.
-	if err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("cannot remove apparmor profile cache: %s", err)
+	for _, name := range names {
+		err = os.Remove(filepath.Join(cacheDir, name))
```

Comment:
note to self: add `osutils.RemoveMany` that opens the directory and does `unlinkat` on the files.

Reply:
Oh, I have a helper that will help with that. I will push it in a second. Unlinkat with flags argument :)